spec/controllers/internal_api/v1/pillars_controller_spec.rb

```diff
@@ -434,97 +434,107 @@
     end
   end
 
-  def expected_dex_json(num, certificate)
-    {
-      id:              num,
-      name:            "LDAP Server #{num}",
-      root_ca_data:    Base64.encode64(certificate.certificate),
-      bind:            {
-        anonymous: false,
-        dn:        "cn=admin,dc=ldap_host_#{num},dc=com",
-        pw:        "pass"
-      },
-      username_prompt: "Username",
-      user:            {
-        base_dn:  "cn=users,dc=ldap_host_#{num},dc=com",
-        filter:   "(objectClass=person)",
-        attr_map: {
-          username: "uid",
-          id:       "uid",
-          email:    "mail",
-          name:     "name"
-        }
-      },
-      group:           {
-        base_dn:  "cn=groups,dc=ldap_host_#{num},dc=com",
-        filter:   "(objectClass=group)",
-        attr_map: {
-          user:  "uid",
-          group: "member",
-          name:  "name"
-        }
-      }
-    }
-  end
-
-  # rubocop:disable RSpec/ExampleLength
-  context "with dex LDAP connectors tls" do
-    it "has dex LDAP connectors" do
-      dex_connector_ldap = create(:dex_connector_ldap, :tls, :regular_admin)
-      CertificateService.create(service: dex_connector_ldap, certificate: certificate)
+  context "with Dex LDAP connectors" do
+    before do |example|
+      # calculate a likely-unique 0-100 integer hash value by adding up the
+      # integer values of chars in the example name (context + example)
+      u = example.metadata[:full_description].codepoints.reduce(0, :+) % 100
+      @conn_name = "LDAP connector #{u}"
+      @host_name = "ldap_host_#{u}.com"
```

Comment:
Why do we need this? The factory is supposed to take care of that as we can see [here](https://github.com/kubic-project/velum/blob/master/spec/factories/dex_connectors_ldap_factory.rb#L3). Is it not working?

Reply:
The attempt here is to avoid using the sequence for the names; there is a conflict where the sequence and db ids don't match up. Really; if this change is done the setting of the host should be removed from the factory and/or a comment added there as that is the purpose of that code being there.